schematic_api/api/security_controller_.py

```diff
@@ -1,14 +1,48 @@
-from typing import List
+import logging
+from typing import Dict, Union
 
+from jwt import PyJWKClient, decode
+from jwt.exceptions import InvalidTokenError
+from synapseclient import Synapse
 
-def info_from_bearerAuth(token):
+from schematic.configuration.configuration import CONFIG
+
+logger = logging.getLogger(__name__)
+
+syn = Synapse(
+    configPath=CONFIG.synapse_configuration_path,
+    cache_client=False,
+)
+jwks_client = PyJWKClient(
+    uri=syn.authEndpoint + "/oauth2/jwks", headers=syn._generate_headers()
+)
+
+
+def info_from_bearerAuth(token: str) -> Dict[str, Union[str, int]]:
     """
-    Check and retrieve authentication information from custom bearer token.
-    Returned value will be passed in 'token_info' parameter of your operation function, if there is one.
-    'sub' or 'uid' will be set in 'user' parameter of your operation function, if there is one.
-    :param token Token provided by Authorization header
-    :type token: str
-    :return: Decoded token information or None if token is invalid
-    :rtype: dict | None
+    Authenticate user using bearer token. The token claims are decoded and returned.
+
+    Example from:
+    <https://pyjwt.readthedocs.io/en/stable/usage.html#retrieve-rsa-signing-keys-from-a-jwks-endpoint>
+
+    Args:
+        token (str): Bearer token.
+
+    Returns:
+        dict: Decoded token information.
     """
-    return {"uid": "user_id"}
+    try:
+        jwks_client.get_signing_key_from_jwt(token)
+        signing_key = jwks_client.get_signing_key_from_jwt(token)
+        data = decode(
+            jwt=token,
+            key=signing_key.key,
+            algorithms=["RS256"],
+            options={"verify_aud": False},
+        )
+
+        return data
+    except InvalidTokenError:
+        logger.exception("Error decoding authentication token")
+        # When the return type is None the web framework will return a 401 OAuthResponseProblem exception
+        return None
```

Comment:
https://github.com/Sage-Bionetworks/schematic/blob/fds-2373-jwt-verification/tests/test_api.py#L130-L138

is testing that an invalid header is returning back the expected HTTP 401 error.

Every other test is handling the other case where valid auth headers are needed.